ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/block/access/list/AccessLocationTracker.java

```diff
@@ -0,0 +1,200 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet.block.access.list;
+
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.StorageSlotKey;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.ethereum.mainnet.block.access.list.PartialBlockAccessView.AccountChangesBuilder;
+import org.hyperledger.besu.ethereum.mainnet.block.access.list.PartialBlockAccessView.PartialBlockAccessViewBuilder;
+import org.hyperledger.besu.evm.account.Account;
+import org.hyperledger.besu.evm.frame.Eip7928AccessList;
+import org.hyperledger.besu.evm.worldstate.StackedUpdater;
+import org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount;
+import org.hyperledger.besu.evm.worldstate.WorldUpdater;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.vertx.core.impl.ConcurrentHashSet;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt256;
+
+public class AccessLocationTracker implements Eip7928AccessList {
+
+  private final int blockAccessIndex;
+  private final Map<Address, AccountAccessList> touchedAccounts = new ConcurrentHashMap<>();
+
+  public AccessLocationTracker(final int blockAccessIndex) {
+    this.blockAccessIndex = blockAccessIndex;
+  }
+
+  @Override
+  public void clear() {
+    touchedAccounts.clear();
+  }
+
+  @Override
+  public void addTouchedAccount(final Address address) {
+    touchedAccounts.putIfAbsent(address, new AccountAccessList(address));
+  }
+
+  @Override
+  public void addSlotAccessForAccount(final Address address, final UInt256 slotKey) {
+    addTouchedAccount(address);
+    touchedAccounts.get(address).addSlotAccess(slotKey);
+  }
+
+  public static final class AccountAccessList {
+    private final Address address;
+    private final Set<UInt256> slots = new ConcurrentHashSet<>();
+
+    public AccountAccessList(final Address address) {
+      this.address = address;
+    }
+
+    public void addSlotAccess(final UInt256 slotKey) {
+      slots.add(slotKey);
+    }
+
+    public Address getAddress() {
+      return address;
+    }
+
+    public Set<UInt256> getSlots() {
+      return slots;
+    }
+
+    @Override
+    public String toString() {
+      return "AccountAccessList{" + "address=" + address + ", slots=" + slots + '}';
+    }
+  }
+
+  public PartialBlockAccessView createPartialBlockAccessView(final WorldUpdater updater) {
```

Comment:
Why _partial_? what is missing to be complete? I think some doc or comments are useful for who is reading this for the first time